http/delete_test.go

```diff
@@ -0,0 +1,355 @@
+				body: fmt.Sprintf(`{
+					"code": "invalid",
+					"message": "Please provide either bucketID or bucket"
+				  }`),
+			},
+		},
+		{
+			name: "insufficient permissions delete",
+			args: args{
+				queryParams: map[string][]string{
+					"org":    []string{"org1"},
+					"bucket": []string{"buck1"},
+				},
+				body:       []byte(`{"start":"2009-01-01T23:00:00Z","stop":"2019-11-10T01:00:00Z"}`),
+				authorizer: &influxdb.Authorization{UserID: user1ID},
+			},
+			fields: fields{
+				BucketService: &mock.BucketService{
+					FindBucketFn: func(ctx context.Context, f influxdb.BucketFilter) (*influxdb.Bucket, error) {
+						return &influxdb.Bucket{
+							ID:   influxdb.ID(2),
+							Name: "bucket1",
+						}, nil
+					},
+				},
+				OrganizationService: &mock.OrganizationService{
+					FindOrganizationF: func(ctx context.Context, f influxdb.OrganizationFilter) (*influxdb.Organization, error) {
+						return &influxdb.Organization{
+							ID: influxdb.ID(1),
+						}, nil
+					},
+				},
+			},
+			wants: wants{
+				statusCode:  http.StatusForbidden,
+				contentType: "application/json; charset=utf-8",
+				body: fmt.Sprintf(`{
+					"code": "forbidden",
+					"message": "insufficient permissions to delete"
+				  }`),
+			},
+		},
+		{
+			name: "no predicate delete",
+			args: args{
+				queryParams: map[string][]string{
+					"org":    []string{"org1"},
+					"bucket": []string{"buck1"},
+				},
+				body: []byte(`{"start":"2009-01-01T23:00:00Z","stop":"2019-11-10T01:00:00Z"}`),
+				authorizer: &influxdb.Authorization{
+					UserID: user1ID,
+					Status: influxdb.Active,
+					Permissions: []influxdb.Permission{
+						{
+							Action: influxdb.WriteAction,
+							Resource: influxdb.Resource{
+								Type:  influxdb.BucketsResourceType,
+								ID:    influxtesting.IDPtr(influxdb.ID(2)),
+								OrgID: influxtesting.IDPtr(influxdb.ID(1)),
+							},
+						},
+					},
+				},
+			},
+			fields: fields{
+				DeleteService: mock.NewDeleteService(),
+				BucketService: &mock.BucketService{
+					FindBucketFn: func(ctx context.Context, f influxdb.BucketFilter) (*influxdb.Bucket, error) {
+						return &influxdb.Bucket{
+							ID:   influxdb.ID(2),
+							Name: "bucket1",
+						}, nil
+					},
+				},
+				OrganizationService: &mock.OrganizationService{
+					FindOrganizationF: func(ctx context.Context, f influxdb.OrganizationFilter) (*influxdb.Organization, error) {
+						return &influxdb.Organization{
+							ID:   influxdb.ID(1),
+							Name: "org1",
+						}, nil
+					},
+				},
+			},
+			wants: wants{
+				statusCode: http.StatusNoContent,
+				body:       fmt.Sprintf(``),
+			},
+		},
+		{
+			name: "complex delete",
+			args: args{
+				queryParams: map[string][]string{
+					"org":    []string{"org1"},
+					"bucket": []string{"buck1"},
+				},
+				body: []byte(`{
+					"start":"2009-01-01T23:00:00Z",
+					"stop":"2019-11-10T01:00:00Z",				
+					"nodeType": "logical",
```

Comment:
I would expect to see a `predicate` key in the top-level object. Instead it looks like the contents of the predicate are mixed with the `start` and `stop` keys.